pkg/volume/flexvolume/flexvolume.go

```diff
@@ -301,8 +301,8 @@ func (f *flexVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 			options = append(options, "rw")
 		}
 		// Extract secret and pass it as options.
-		if f.secret != "" {
-			options = append(options, "secret="+f.secret)
+		for name, secret := range f.secrets {
+			f.options[optionKeySecret+"/"+name] = secret
```

Comment:
Password content is base64 encoded in the secret definition to allow passage of binary data. It makes sense, therefore, for the password to be base64 encoded between this code and the script since it is marshalled to JSON. I realize that would be a breaking change (since flex doesn't do that with existing passwords) though, and could be persuaded it is not a good idea.